engine/lib/filestore.php

```diff
@@ -517,6 +517,11 @@ function _elgg_filestore_detect_mimetype($hook, $type, $mime_type, $params) {
 	if ($mime_type == "application/vnd.ms-office" && $info['extension'] == "ppt") {
 		$mime_type = "application/vnd.ms-powerpoint";
 	}
+	
+	// hack for browser providing better mime-type than server can detect
+	if ($mime_type == "application/octet-stream" && preg_match('~^(audio|image|video)/~', $params['default'])) {
```

Comment:
that would mean we have to maintain a extension mapping and also means trusting files to have the right extension. This fix is catching those cases where the server only detects octet-stream and then checks if browser thought it was audio/image or video. Only then it will use the browser mimetype